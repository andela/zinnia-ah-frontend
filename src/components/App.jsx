import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';

import './App.scss';

class App extends Component {
  render() {
    return (
      <div>
        <ToastContainer autoClose={5000} />
        <h1 className="toothy"> My React App!! </h1>
        <Link to="/login">
          <button>Login</button>
        </Link>
        <Link to="/signup">
          <button>Signup</button>
        </Link>
      </div>
    );
  }
}

export default App;
