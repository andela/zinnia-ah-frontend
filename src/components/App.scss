@import '../assets/variables';

.toothy {
  color: $primary-black;
}

.pt-3 {
  padding-top: 3.125rem;
}

.nav-items {
  display: flex;

  .search-bar {
    margin-right: 1.25rem;
  }
}

.search-bar {
  height: 3.125rem;
  width: 22.9375rem;

  .left {
    margin-top: 0;
    padding-right: 0;
  }
}

@media only screen and (max-width: 37.5rem) {

  .pt-3 {
    padding-top: .5rem;
  }
}

@media screen and (min-width: 37.5rem) and ( max-width: 64rem) {
  .pt-3 {
    padding-top: .5rem;
  }
}


@media only screen and (max-width: 28.625rem) {
  .pt-3 {
    padding-top: .5rem;
  }
}
