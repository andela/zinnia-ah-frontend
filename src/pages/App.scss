@import '../../node_modules/@glidejs/glide/src/assets/sass/glide.core';
@import '../assets/variables';

.pt-3 {
  padding-top: 3.125rem;
}

.nav-items {
  display: flex;

  .search-bar {
    margin-right: 1.25rem;
  }
}

.search-bar {
  height: 3.125rem;
  width: 22.9375rem;

  .left {
    margin-top: 0;
    padding-right: 0;
  }

  input[type='text'] {
    height: 3.125rem;
    width: 22.9375rem;
  }
}

.top {
  display: flex;
  margin-top: 4.875rem;
  padding: 0 6%;

  .trending {
    width: 80%;
  }

  .trending-inner {
    display: flex;
  }

  .trending-inner.up {
    margin-bottom: 4.6875rem;
  }

  .popular {
    width: 20%;
  }
}

.justify-content-center {
  justify-content: center;
}

.v-card-container {
  height: content-box;
  padding-right: 6%;
  width: 44%;

  .ui.card img.avatar {
    height: 3.125rem;
    width: 3.125rem;
  }

  .ui.card > .card-footer {
    border-top: 0;
    box-shadow: none;
    flex-grow: 0;
    left: 0;
    margin: 0;
    max-width: 100%;
    min-height: 0;
    padding: 0 1rem;
    position: static;
    top: 0;
    transition: color .1s ease;
    width: auto;
  }

  .author-slot {
    margin-top: 0;
  }
}

.h-card-container {
  padding-right: 6%;
  width: 56%;
}

.ads {
  display: flex;
}

.middle {
  display: flex;
  margin: 3.75rem 0 0;
  padding: 0 6%;

  .tag-result-div {
    border-bottom: .0625rem solid $secondary-grey;
    border-top: .0625rem solid $secondary-grey;
    display: flex;
    margin-right: auto;
    padding: 3.125rem 0;
    width: 76%;
  }

  .h-card-container {
    padding-right: 0;
    width: 56%;
  }

  .h-card-container:first-child {
    padding-right: 3%;
  }

  .tag-div {
    border-bottom: .0625rem solid $secondary-grey;
    border-top: .0625rem solid $secondary-grey;
    padding: 1.25rem 0;
    width: 20%;
  }

  .tag-div .inner {
    display: flex;
    flex-wrap: wrap;
  }
}

.main-bottom {
  display: flex;
  margin: 0 0 1rem;
  padding: 0 6%;

  .suggested-reads {
    display: block;
    margin-right: auto;
    padding: 1.875rem 0 0;
    width: 76%;
  }

  .footer {
    padding: 1.25rem 0 0;
    width: 20%;
  }

  .footer-link {
    color: $primary-black-opacity-6;
    padding-right: .625rem;
  }
}

.glide__slide--active .h-card {
  box-shadow: 0 4px 4px $primary-black-opacity-2;
}

.glide__bullet {
  background: $secondary-grey;
  border: 0;
  border-radius: 100%;
  height: 1.5rem;
  margin: 0 1rem;
  width: 1.5rem;
}

.glide__bullet--active {
  background: $primary-black;
}

.glide__bullet:focus,
.glide__bullet--active:focus {
  outline: none;
}

.glide__bullets {
  display: flex;
  justify-content: center;
}

@media only screen and (max-width: 37.5rem) {

  .pt-3 {
    padding-top: .5rem;
  }

  .top {
    display: block;
    margin-top: 0;
    padding: 0 6%;

    .trending {
      width: 100%;
    }

    .trending-inner {
      display: block;
    }

    .trending-inner.up {
      margin-bottom: 1.75rem;
    }

    .trending-inner .v-card-container {
      width: 100%;
    }

    .v-card-container {
      padding-right: 0%;

      .ui.card img.avatar {
        height: 2.125rem;
        width: 2.125rem;
      }

      .ui.card > .card-footer {
        padding: 0;
      }

      .ui.card > .content {
        padding: 1em 0;
      }
    }


    .trending-inner .h-card-container {
      display: none;
    }

    .popular {
      margin-top: 2rem;
      width: 100%;
    }
  }

  .author-slot-container {
    display: flex;
    overflow-x: auto;

    .author-slot {
      border: .0625rem solid $primary-black-opacity-1;
      border-radius: .125rem;
      box-sizing: border-box;
      margin-right: 1rem;
      padding: 0 .5rem;
    }

    .user-thumbnail {
      margin-right: 2.5rem;
    }
  }

  .author-slot-container::-webkit-scrollbar {
    display: none;
  }

  .index-title {
    text-align: center;
  }

  .ads {
    display: none;
  }

  .middle {
    display: flex;
    flex-direction: column;
    margin: 3.75rem 0;

    .tag-result-div {
      border-bottom: .0625rem solid $secondary-grey;
      border-top: .0625rem solid $secondary-grey;
      display: block;
      margin-right: auto;
      order: 2;
      padding: 3.125rem 0;
      width: 100%;
    }

    .h-card-container {
      padding-right: 0;
      width: 100%;
    }

    .h-card-container:first-child {
      padding-right: 0%;
    }

    .h-card {
      margin-bottom: 2.8125rem;
    }

    .tag-div {
      border-bottom: 0;
      border-top: .0625rem solid $secondary-grey;
      order: 1;
      padding: 1.25rem 0 0;
      width: 100%;
    }

    .tag-div .inner {
      display: flex;
      flex-wrap: nowrap;
      overflow-x: auto;
    }

    .tag-div .inner::-webkit-scrollbar {
      display: none;
    }

    .tag-div .inner .tag {
      min-width: fit-content;
    }
  }
}


@media screen and (min-width: 37.5rem) and ( max-width: 64rem) {
  .pt-3 {
    padding-top: .5rem;
  }

  .top {
    display: block;
    margin-top: 0;
    padding: 0 0 0 6%;

    .trending {
      width: 100%;
    }

    .trending-inner {
      display: flex;
    }

    .trending-inner.up {
      margin-bottom: 3.75rem;
    }

    .trending-inner .v-card-container {
      width: 44%;
    }

    .v-card-container {
      padding-right: 6%;

      .ui.card img.avatar {
        height: 2.125rem;
        width: 2.125rem;
      }

      .ui.card > .card-footer {
        padding: 0;
      }

      .ui.card > .content {
        padding: 1em 0;
      }
    }


    .trending-inner .h-card-container {
      display: block;
      width: 56%;
    }

    .popular {
      margin-top: 2rem;
      width: 100%;
    }
  }

  .author-slot-container {
    display: flex;
    overflow-x: auto;

    .author-slot {
      border: .0625rem solid $primary-black-opacity-1;
      border-radius: .125rem;
      box-sizing: border-box;
      margin-right: 1rem;
      padding: 0 .5rem;
    }

    .user-thumbnail {
      margin-right: 2.5rem;
    }
  }

  .author-slot-container::-webkit-scrollbar {
    display: none;
  }

  .index-title {
    text-align: center;
  }

  .ads {
    display: none;
  }


  .middle {
    display: flex;
    flex-direction: column;
    margin: 3.75rem 0;

    .tag-result-div {
      border-bottom: .0625rem solid $secondary-grey;
      border-top: .0625rem solid $secondary-grey;
      display: block;
      margin-right: auto;
      order: 2;
      padding: 3.125rem 0;
      width: 100%;
    }

    .h-card-container {
      padding-right: 0;
      width: 100%;
    }

    .h-card-container:first-child {
      padding-right: 0%;
    }

    .h-card {
      margin-bottom: 2.8125rem;
    }

    .tag-div {
      border-bottom: 0;
      border-top: .0625rem solid $secondary-grey;
      order: 1;
      padding: 1.25rem 0 0;
      width: 100%;
    }

    .tag-div .inner {
      display: flex;
      flex-wrap: nowrap;
      overflow-x: auto;
    }

    .tag-div .inner::-webkit-scrollbar {
      display: none;
    }

    .tag-div .inner .tag {
      min-width: fit-content;
    }
  }
}


@media only screen and (max-width: 28.625rem) {

  .pt-3 {
    padding-top: .5rem;
  }

  .top {
    display: block;
    margin-top: 0;
    padding: 0 6%;

    .trending {
      width: 100%;
    }

    .trending-inner {
      display: block;
    }

    .trending-inner.up {
      margin-bottom: 1.75rem;
    }

    .trending-inner .v-card-container {
      width: 100%;
    }

    .v-card-container {
      padding-right: 0%;

      .ui.card img.avatar {
        height: 2.125rem;
        width: 2.125rem;
      }

      .ui.card > .card-footer {
        padding: 0;
      }

      .ui.card > .content {
        padding: 1em 0;
      }
    }


    .trending-inner .h-card-container {
      display: none;
    }

    .popular {
      margin-top: 2rem;
      width: 100%;
    }
  }

  .author-slot-container {
    display: flex;
    overflow-x: auto;

    .author-slot {
      border: .0625rem solid $primary-black-opacity-1;
      border-radius: .125rem;
      box-sizing: border-box;
      margin-right: 1rem;
      padding: 0 .5rem;
    }

    .user-thumbnail {
      margin-right: 2.5rem;
    }
  }

  .author-slot-container::-webkit-scrollbar {
    display: none;
  }

  .index-title {
    text-align: center;
  }

  .ads {
    display: none;
  }

  .middle {
    display: flex;
    flex-direction: column;
    margin: 3.75rem 0;

    .tag-result-div {
      border-bottom: .0625rem solid $secondary-grey;
      border-top: .0625rem solid $secondary-grey;
      display: block;
      margin-right: auto;
      order: 2;
      padding: 3.125rem 0;
      width: 100%;
    }

    .h-card-container {
      padding-right: 0;
      width: 100%;
    }

    .h-card-container:first-child {
      padding-right: 0%;
    }

    .h-card {
      margin-bottom: 2.8125rem;
    }

    .tag-div {
      border-bottom: 0;
      border-top: .0625rem solid $secondary-grey;
      order: 1;
      padding: 1.25rem 0 0;
      width: 100%;
    }

    .tag-div .inner {
      display: flex;
      flex-wrap: nowrap;
      overflow-x: auto;
    }

    .tag-div .inner::-webkit-scrollbar {
      display: none;
    }

    .tag-div .inner .tag {
      min-width: fit-content;
    }
  }

  .main-bottom {
    display: block;
    margin: 0;
    padding: 0 6%;

    .suggested-reads {
      display: block;
      margin-right: auto;
      padding: 1.875rem 0 0;
      width: 100%;
    }

    .footer {
      padding: 1.25rem 0 0;
      width: 100%;
    }

    .footer-link {
      display: none;
    }
  }
}


