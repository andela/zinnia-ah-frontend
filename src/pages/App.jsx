import React, { Component, Fragment } from 'react';
import { Link } from 'react-router-dom';
import { Input } from 'semantic-ui-react';
import 'react-toastify/dist/ReactToastify.css';

// components
import Navbar from '../components/presentationals/Navbar/Navbar';
import Button from '../components/presentationals/Button/Button';
import PopularAuthorsList from '../components/presentationals/PopularAuthorsList/PopularAuthorsList';
import Title from '../components/presentationals/Title/Title';
import VerticalCard from '../components/presentationals/VerticalCard/VerticalCard';

// styles
import './App.scss';
import HorizontalCard from '../components/presentationals/HorizontalCard/HorizontalCard';
import Image from '../components/presentationals/Image/Image';

class App extends Component {
  state = {
    authors: [
      {
        id: 23,
        name: 'Nedy',
        url: 'https://google.com',
        image:
          'https://res.cloudinary.com/nedy123/image/upload/v1544002134/xykn3yriqcsgrbwrqwa6.jpg',
        username: 'nedyudombat',
      },
      {
        id: 24,
        name: 'Samantha',
        url: 'https://google.com',
        image:
          'https://res.cloudinary.com/nedy123/image/upload/v1516819339/lady-2_jqx4go.jpg',
        username: 'samanthress',
      },
      {
        id: 25,
        name: 'Hilary',
        url: 'https://google.com',
        image:
          'https://res.cloudinary.com/nedy123/image/upload/v1516819339/dummy-av.jpg',
        username: 'noMoreEben',
      },
    ],
  };
  Searchbar() {
    return (
      <form className="search-bar">
        <Input icon="search" iconPosition="left" placeholder="Search..." />
      </form>
    );
  }

  render() {
    return (
      <Fragment>
        <div className="pt-3">
          <Navbar>
            <div className="nav-items">
              {this.Searchbar()}
              <Link to="/login">
                <Button type="submit" value="WRITE" className="btn-dark w-10" />
              </Link>
            </div>
          </Navbar>
        </div>
        <div className="top">
          <div className="trending">
            <Title content="Trending" className="title-md index-title" />
            <div className="trending-inner up">
              <div className="v-card-container">
                <VerticalCard />
              </div>
              <div className="h-card-container">
                <HorizontalCard />
                <HorizontalCard />
              </div>
            </div>
            <div className="trending-inner bottom">
              <div className="h-card-container">
                <HorizontalCard />
                <HorizontalCard />
              </div>
              <div className="v-card-container">
                <VerticalCard />
              </div>
            </div>
          </div>
          <div className="popular">
            <PopularAuthorsList authors={this.state.authors} />
            <div className="justify-content-center ads">
              <div
                style={{
                  maxWidth: '100%',
                  marginTop: '2rem',
                  position: 'relative',
                }}
              >
                <Image
                  src="https://res.cloudinary.com/nedy123/image/upload/v1558511972/jumia-ad_hvrkw8.png"
                  style={{
                    maxWidth: '100%',
                    position: 'relative',
                    zIndex: '2',
                  }}
                />
                <Image
                  src="https://res.cloudinary.com/nedy123/image/upload/v1558511973/lady-ad_jc3tth.png"
                  style={{
                    maxWidth: '100%',
                    marginTop: '-2.8125rem',
                    position: 'relative',
                    zIndex: '1',
                  }}
                />
                <Image
                  src="https://res.cloudinary.com/nedy123/image/upload/v1558511972/konga-ad_cw2mpn.png"
                  style={{
                    maxWidth: '100%',
                    marginTop: '-3.2625rem',
                    position: 'relative',
                    zIndex: '2',
                  }}
                />
              </div>
            </div>
          </div>
        </div>
      </Fragment>
    );
  }
}

export default App;
