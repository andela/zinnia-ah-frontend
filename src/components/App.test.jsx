import React from 'react';
import { shallow } from 'enzyme';
// import { BrowserRouter as Router } from 'react-router-dom';

import App from './App';

describe('App', () => {
  it('should render correctly', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });
});
