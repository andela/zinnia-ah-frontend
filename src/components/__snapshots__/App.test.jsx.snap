// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div>
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    progressClassName={null}
    progressStyle={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
  <h1
    className="toothy"
  >
     My React App!! 
  </h1>
  <Link
    to="/login"
  >
    <button>
      Login
    </button>
  </Link>
  <Link
    to="/signup"
  >
    <button>
      Signup
    </button>
  </Link>
</div>
`;
